src/components/astra-editor/plugins/prompt.ts

```diff
@@ -153,62 +143,20 @@ class PromptWidget extends WidgetType {
         effects: [effectClearHighlight.of(null)],
       })
 
-      input.focus()
-      input.select()
-    }
-
-    rejectButton.addEventListener('click', rejectQuery)
-    generateButton.addEventListener('click', generateQuery)
-
-    input.addEventListener('mousedown', (e) => {
-      e.stopPropagation()
+      container.showReject = true
     })
 
-    input.addEventListener('paste', (e) => {
-      e.stopPropagation()
-    })
-
-    input.addEventListener('copy', (e) => {
-      e.stopPropagation()
-    })
-
-    input.addEventListener('keydown', (e) => {
-      if (e.key === 'Escape') {
-        view.dispatch({ effects: [effectHidePrompt.of(null)] })
-        setTimeout(() => view.focus(), 50)
-        e.preventDefault()
-      } else if ((e.metaKey || e.ctrlKey) && e.key === 'Enter') {
-        generateQuery()
-        e.preventDefault()
-      } else if ((e.metaKey || e.ctrlKey) && e.key === 'z') {
-        rejectQuery()
-        e.preventDefault()
-      }
-
-      e.stopPropagation()
-    })
-
-    input.addEventListener('keyup', (e) => {
-      // Adjust the height input
-      fakeInput.value = input.value
-      input.style.height = Math.max(fakeInput.scrollHeight, 50) + 'px'
-      e.stopPropagation()
-    })
-
-    input.addEventListener('keypress', (e) => {
-      e.stopPropagation()
-    })
-
-    setTimeout(() => input.focus(), 20)
-
-    container.addEventListener('mousedown', (e) => {
-      e.stopPropagation()
-    })
+    setTimeout(() => {
+      container.focus()
+    }, 50)
```

Comment:
`setTimeout`s always give me pause 😏 but seriously would `0` also work? or does it need to be many milliseconds for some reason? is there an event we could react to instead?

Reply:
Not sure if 0 will work. I will test it out. But I don't think human can react in 50ms second. The reason we use `setTimeout` because autofocus property does not work. Will add more comment. Sorry about that

Reply:
the `0` just lets it schedule on the next event loop, which is often sufficient, but it depends of course